5.1

ortalyatzkan opened the pull request.

ortalyatzkan committed:
add exercise file

ortalyatzkan committed:
change name of function

ortalyatzkan committed:
change

ortalyatzkan committed:
add i_dont_have_vinaigrette

ortalyatzkan committed:
add This_is_the_path.py

ortalyatzkan committed:
delete exercise.py file

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+import random
+import calendar
+import datetime
+
```
two lines after imports

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+    random_date = get_random_date_in_range(s_date, e_date)
+    print(random_date)
+    if get_if_is_tuesday(random_date): print("I don't have vinaigrette!")
+
```
two lines between functions

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+    """
+    random_date = get_random_date_in_range(s_date, e_date)
+    print(random_date)
+    if get_if_is_tuesday(random_date): print("I don't have vinaigrette!")
```
Dont do single line ifs  unless its a ternary operator - https://stackoverflow.com/questions/2802726/putting-a-simple-if-then-else-statement-on-one-line

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+
+
+
+def  print_random_date(s_date, e_date):
```
extra space here

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+def  print_random_date(s_date, e_date):
+    """
+    A function that prints a random date in range and if the date Monday prints I do not have a vinaigrette
+    :param s_date: s_date:start date (format YYYY-MM-DD)
```
e_date and s_date are bad names-  dont use abbreviations - start date and end date -

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+    random_date = s_date + datetime.timedelta(days=random_number_of_days)
+    return random_date
+
+def get_if_is_tuesday(random_date):
```
ון שאני הולך למכולת רק **בימי שני** ואני צרכן כבד של רוטב ויניגרט, אם התאריך נופל על **יום שני**

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+
+
+
+def  print_random_date(s_date, e_date):
```
Add type hints and return types - read here about how its done - https://docs.python.org/3/library/typing.html

ortalyatzkan replied:
datetime.date can be written as a variable type and return type?

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+
+
+def  print_random_date(s_date, e_date):
+    """
```
your funciton should look this -
![Capture](https://user-images.githubusercontent.com/13467749/160145323-82ad31f5-a551-472e-8195-7355cb7c354a.JPG)

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+def get_if_is_tuesday(random_date):
+    """
+    check if the random day is Monday
+    :param random_date:
```
Why isnt this parameter documented?

talbarfm commented on I_dont_have_vinaigrette.py:
```diff
@@ -0,0 +1,43 @@
+
+def get_if_is_tuesday(random_date):
+    """
+    check if the random day is Monday
```
Start sentences with capital letter, finish with a .

talbarfm commented on This_is_the_path.py:
```diff
@@ -0,0 +1,14 @@
+
+if __name__== '__main__':
+    path = 'C:\\Users\\user\\Desktop\\ortalYatzkan'
+    print(list_dir_start_with_deep(path))
```
Add newline in eof

talbarfm commented on This_is_the_path.py:
```diff
@@ -0,0 +1,14 @@
+
+#a function that returns the name of all directory that start with deep in the path he get
+def list_dir_start_with_deep(path):
+     return [ file for file in listdir(path) if file.startswith('deep') and os.path.isdir(os.path.join(path, file))]
```
space after [ is not needed

talbarfm commented on This_is_the_path.py:
```diff
@@ -0,0 +1,14 @@
+from os.path import join, isfile
+import pathlib
+
+#a function that returns the name of all directory that start with deep in the path he get
```
Document in the right format

talbarfm commented on This_is_the_path.py:
```diff
@@ -0,0 +1,14 @@
+#a function that returns the name of all directory that start with deep in the path he get
+def list_dir_start_with_deep(path):
+     return [ file for file in listdir(path) if file.startswith('deep') and os.path.isdir(os.path.join(path, file))]
+
```
Thats a redundant newline

ortalyatzkan committed:
Corrected errors in I_dont_have_vinaigrette.py

ortalyatzkan committed:
Corrected errors in This_is_the_path.py

ortalyatzkan committed:
Add type hints and return types in I_dont_have_vinaigrette.py
nts and return types# Please enter the commit message for your changes. Lines starting

lovodkin93 commented on I_dont_have_vinaigrette.py:
```diff
@@ -3,41 +3,42 @@
-    :param e_date: e_date:end date (format YYYY-MM-DD)
+    A function that prints a random date in range and if the date Monday prints :"I do not have a vinaigrette".
+    :param start_date: start_date:Start date of the range(format YYYY-MM-DD).
+    :param end_date: end_date:End date of the range (format YYYY-MM-DD).
```
missing @return: None in the documentation.

ortalyatzkan committed:
Add return in function documentation
